src/utils/registerNewNetwork.ts

```diff
@@ -5,11 +5,88 @@ import {
   registerCustomArbitrumNetwork,
 } from '@arbitrum/sdk';
 import { testnets } from '../chains';
+import {
+  createRollupFetchTransactionHash,
+  getRollupInitializedEvents,
+} from '../createRollupFetchTransactionHash';
+import { Address, Chain, PublicClient, Transport } from 'viem';
+import { publicClientToProvider } from '../ethers-compat/publicClientToProvider';
+import { createRollupPrepareTransactionReceipt } from '../createRollupPrepareTransactionReceipt';
+import { createTokenBridgeFetchTokenBridgeContracts } from '../createTokenBridgeFetchTokenBridgeContracts';
 
 const isTestnet = (parentChainId: number) => {
   return testnets.some((testnet) => testnet.id === parentChainId);
 };
 
+export async function prepareRegisterNewNetworkParams<TChain extends Chain | undefined>({
+  parentChainPublicClient,
+  rollupAddress,
+}: {
+  parentChainPublicClient: PublicClient<Transport, TChain>;
+  rollupAddress: Address;
+}): Promise<ArbitrumNetwork> {
```

Comment:
let's make a few adjustments to be consistent with other APIs + add jsdoc

```suggestion
export async function prepareArbitrumNetwork<TChain extends Chain | undefined>(
  publicClient: PublicClient<Transport, TChain>,
  { rollup }: { rollup: Address },
): Promise<ArbitrumNetwork> {
```